File: frontend/fashion_advisor_ai/lib/models/clothing.dart
class Clothing {
  final String id;
  final String title;
  final String description;
  final String imageUrl;
  final String brand;
  final String priceRange;
  final List<String> styleTags;
  final List<String> colors;
  final List<String> items;
  final String mainCategory;
final String targetAudience;

  Clothing({
    required this.id,
    required this.title,
    required this.description,
    required this.imageUrl,
    required this.brand,
    required this.priceRange,
    required this.styleTags,
    required this.colors,
    required this.items,
    required this.mainCategory,
    required this.targetAudience,
  });

  factory Clothing.fromJson(Map<String, dynamic> json) {
    return Clothing(
      id: json["_id"] ?? "",
      title: json["title"] ?? "",
      description: json["description"] ?? "",
      imageUrl: json["imageUrl"] ?? "",
      brand: json["brand"] ?? "",
      priceRange: json["priceRange"] ?? "",
      styleTags: List<String>.from(json["styleTags"] ?? []),
      colors: List<String>.from(json["colors"] ?? []),
      items: List<String>.from(json["items"] ?? []),
      mainCategory: json["mainCategory"] ?? "",
      targetAudience: json["targetAudience"] ?? "",
    );
  }
}

File: backend/models/Clothing.js
const mongoose = require("mongoose");

const ClothingSchema = new mongoose.Schema({

    title: {
        type: String,
        required: true,
    },

    description: {
        type: String,
    },

    imageUrl: {
        type: String,
        required: true,
    },

    imageType: {
        type: String,
        enum: [
            "Full Outfit",
            "Single Item",
            "Street Style",
            "Editorial",
            "Flatlay"
        ],
    },

    targetAudience: {
        type: String,
        enum: [
            "Women",
            "Men",
            "Unisex"
        ],
    },

    mainCategory: {
        type: String,
        enum: [
            "Full Outfit",
            "Top",
            "Bottom",
            "Dress",
            "Outerwear",
            "Footwear",
            "Accessory"
        ],
    },

    items: {
        type: [String],
        default: [],
    },

    styleTags: {
        type: [String],
        default: [],
    },

    colors: {
        type: [String],
        default: [],
    },

    pattern: String,

    fabric: {
        type: [String],
        default: [],
    },

    fit: String,

    sleeveLength: String,

    neckline: String,

    bottomLength: String,

    occasion: {
        type: [String],
        default: [],
    },

    season: {
        type: [String],
        default: [],
    },

    weather: {
        type: [String],
        default: [],
    },

    brand: String,

    priceRange: {
    type: String,
    enum: [
          "Unknown",
        "Budget",
        "Mid",
        "Premium",
        "Luxury"
    ]
},

    searchCount: {
        type: Number,
        default: 0,
    },

    viewCount: {
        type: Number,
        default: 0,
    },

    saveCount: {
        type: Number,
        default: 0,
    },

    clickCount: {
        type: Number,
        default: 0,
    },

    similarOutfits: {
        type: [String],
        default: [],
    }

},
{
    timestamps: true,
});

module.exports = mongoose.model(
    "Clothing",
    ClothingSchema
);